pkg/modprovider/server.go

```diff
@@ -323,6 +328,18 @@ func (s *server) Configure(
 
 	s.providerConfig = config
 
+	if config.HasValue(resource.PropertyKey(moduleExecutorVariableName)) {
+		if executor, ok := config[moduleExecutorVariableName]; ok && executor.IsString() {
+			// remove quotes from the string value in case it was JSON-encoded
```

Comment:
Question, why do you think the string is going to ever be JSON-encoded? IF you are worried about legacy JSON-encodings then AFAIK it never affects stringly typed properties.

Reply:
I thought all args would be individually JSON-encoded but now double checking the generated code for TypeScript
```ts
resourceInputs["executor"] = (args ? args.executor : undefined) ?? (utilities.getEnv("PULUMI_TERRAFORM_MODULE_EXECUTOR") || "");
```
doesn't look like it. I can simplify